### ISSUE TITLE
[Bug]: Window doesn't resize properly on XWayland

### ISSUE THREAD

**Issue by ghost** [brndnmtthws/conky#2045]
### What happened?

Running labwc, I have the sfwbar as taskbar. When I kill the taskbar the conky window resizes to a small square in the middle of the screen, just showing the top output. The only way to fix this is to set "own_window_type = override", but this means that the output text will overlap with any other window that I will open.

### Version

1.21.6

### Which OS/distro are you seeing the problem on?

Arch Linux

**Comment by Bluey26** [comment]
Hi, i do not use sfwbar but it also happens when i kill my panel(lxqt-panel).

To fix this, you can kill and re-start conky, i use the following in a .desktop to do it:

`Exec=bash -c 'pkill conky && conky &>/dev/null'`

I'm in labwc too.

**Comment by Consolatis** [comment]
As `own_window_type` changes the behavior, this is likely an issue with the xwayland implementation in labwc in combination with the X11 backend of conky. So this doesn't look like a conky wayland backend issue.

**Comment by wastlnd** [comment]
@Consolatis yet I tried conky on sway and I had a similar issue, the window could not resize itself property.

**Comment by Bluey26** [comment]
When using the wayland backend (after #1960 fix), the issue seems to be fixed, no more middle-screen box with conky information when the lxqt-panel is restarted.

**Comment by wastlnd** [comment]
I updated to the latest commit, now the window is unaffected by the taskbar, but I have a new issue: borders show up though they are disabled, and above all the entries get more and more messy, as if words were overlapping
![IMG_20241211_235944_HDR](https://github.com/user-attachments/assets/1c06866b-a452-48e3-986d-ec5d4cc5421f)

**Comment by Bluey26** [comment]
I suspect that the word overlap is 2 conky instances running at the same time. Try `pkill conky && conky`

**Comment by wastlnd** [comment]
@Bluey26 that's what I suspected too but after killing and restarting I get the same behavior. What's really weird is that after a few seconds the issue gets worse, I end up having white bars instead of words.

**Comment by Caellian** [comment]
> I updated to the latest commit, now the window is unaffected by the taskbar, but I have a new issue: borders show up though they are disabled, and above all the entries get more and more messy, as if words were overlapping !

Thanks for reporting this, I believe I broke it in #2105.

**Comment by wastlnd** [comment]
Thanks to you @Caellian , I'll wait for the fix and report

**Comment by Caellian** [comment]
It's likey @bl4kraven fixed this issue in #2127 - I used [`cairo_rectangle`](https://coolbutuseless.github.io/package/cairocore/reference/cairo_rectangle.html) which only clears borders, but #2127 adds `cairo_fill` which should properly clear the entire conky window and prevent text overdraw.

Try out the latest release and let me know whether the text overdraw is fixed.

The main issue still remains open though.

**Comment by wastlnd** [comment]
Hey @Caellian yes, it works now, at least the issues that I was experiencing have completely gone away.
